Resources/Prototypes/Nuclear14/Entities/Objects/Weapons/Guns/Heavy/minigun.yml

```diff
@@ -0,0 +1,113 @@
+- type: entity
+  name: minigun
+  parent: BaseWeaponRifle
+  id: N14WeaponMinigun
+  description: A extremely heavy minigun with 3 barrels, it uses 5mm ammo. # TODO: Need to create projectiles, magazines and boxes for 5.56.
+  components:
+  - type: Tag
+    tags:
+     - N14HeavyGun
+  - type: Sprite
+    sprite: Nuclear14/Objects/Weapons/Guns/Heavy/minigun.rsi
+    layers:
+      - state: base
+        map: ["enum.GunVisualLayers.Base"]
+      - state: mag-0
+        map: ["enum.GunVisualLayers.Mag"]
+  - type: Item
+    sprite: Nuclear14/Objects/Weapons/Guns/Heavy/minigun.rsi
+  - type: Gun
+    minAngle: 6
+    maxAngle: 8  
+    fireRate: 12
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/minigun.ogg
+  - type: ItemSlots
+    slots:
+      gun_magazine:
+        name: Magazine
+        startingItem: N14MinigunMagazine5mm
+        insertSound: /Audio/Weapons/Guns/MagIn/ltrifle_magin.ogg
+        ejectSound: /Audio/Weapons/Guns/MagOut/ltrifle_magout.ogg
+        priority: 2
+        whitelist:
+          tags:
+            - N14MinigunMagazine5mm
+      gun_chamber:
+        name: Chamber
+        startingItem: N14CartridgePistol5
+        priority: 1
+        whitelist:
+          tags:
+            - N14CartridgePistol5
+  - type: ContainerContainer
+    containers:
+      gun_magazine: !type:ContainerSlot
+      gun_chamber: !type:ContainerSlot
+  - type: MagazineVisuals
+    magState: mag
+    steps: 1
+    zeroVisible: true
+  - type: Appearance
+  - type: MultiHandedItem
+  - type: HeldSpeedModifier   
+  - type: ClothingSpeedModifier
+    walkModifier: 0.7
+    sprintModifier: 0.7
+
+- type: entity
+  name: avenger minigun
+  parent: BaseWeaponRifle
```

Comment:
Should probably make a heavy weapon base to cut down a little on code duplication. Should narrow down scope for an error to occur too.